### DISCUSSION TITLE
Recipes not being applied

### DISCUSSION THREAD

**Discussion by evanchooly** [openrewrite/rewrite#4073] [General] [answered]
I have this recipe definition with 3 different recipes in the list and only one gets applied.  the recipe definition can be found [here](https://github.com/MorphiaOrg/morphia/blob/master/rewrite/src/main/resources/META-INF/rewrite/morphia3.yml).  I'm applying this recipe in this test repo via [this](https://github.com/MorphiaOrg/morphia-upgrade-testing/blob/main/.github/test-upgrade.sh#L18) script.  I can see the `ChangeMethodName` being applied but not the `ChangeType` recipe.  You can see [here](https://github.com/MorphiaOrg/morphia-upgrade-testing/actions/runs/8134949770/job/22228631696#step:4:232) that only one file is updated where there should be at least 2.  Can anyone see what I'm doing wrong?

**Comment by timtebeek** [comment, marked as answer]
hi @evanchooly , good seeing you here. I checked out your project to give things a go here in a slightly different way, but with better results.

1. `gh repo clone MorphiaOrg/morphia-upgrade-testing`
2. create a `rewrite.yml` in the root of that project, using the [content taken from here](https://github.com/MorphiaOrg/morphia/blob/master/rewrite/src/main/resources/META-INF/rewrite/morphia3.yml)
3. `mvn  -U org.openrewrite.maven:rewrite-maven-plugin:run -Drewrite.activeRecipes=dev.morphia.UpgradeToMorphia30`, from the root of the project.

That at least gets me the following out put on your sample project:
```
[INFO] --- rewrite:5.23.3:run (default-cli) @ java ---
[INFO] Using active recipe(s) [dev.morphia.UpgradeToMorphia30]
[INFO] Using active styles(s) []
[INFO] Validating active recipes...
[INFO] Project [morphia-upgrade-testing] Resolving Poms...
[INFO] Project [morphia-upgrade-testing] Parsing source files
[INFO] Project [java] Parsing source files
[INFO] Running recipe(s)...
[WARNING] Changes have been made to java/src/test/java/dev/morphia/test/TestBase.java by:
[WARNING]     dev.morphia.UpgradeToMorphia30
[WARNING]         org.openrewrite.java.ChangeType: {oldFullyQualifiedTypeName=dev.morphia.DatastoreImpl, newFullyQualifiedTypeName=dev.morphia.MorphiaDatastore}
[WARNING] Changes have been made to java/src/test/java/dev/morphia/test/aggregation/stages/TestOut.java by:
[WARNING]     dev.morphia.UpgradeToMorphia30
[WARNING]         org.openrewrite.java.ChangeMethodName: {methodPattern=dev.morphia.aggregation.stages.Out to(..), newMethodName=out, matchOverrides=true, ignoreDefinition=false}
[WARNING]         org.openrewrite.java.ChangeType: {oldFullyQualifiedTypeName=dev.morphia.DatastoreImpl, newFullyQualifiedTypeName=dev.morphia.MorphiaDatastore}
[WARNING] Changes have been made to java/src/test/java/dev/morphia/test/MorphiaContainer.java by:
[WARNING]     dev.morphia.UpgradeToMorphia30
[WARNING]         org.openrewrite.java.ChangeType: {oldFullyQualifiedTypeName=dev.morphia.DatastoreImpl, newFullyQualifiedTypeName=dev.morphia.MorphiaDatastore}
[WARNING] Please review and commit the results.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for morphia-upgrade-testing 1.0.0-SNAPSHOT:
[INFO] 
[INFO] morphia-upgrade-testing ............................ SUCCESS [  1.642 s]
[INFO] java ............................................... SUCCESS [  7.670 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  9.804 s
[INFO] Finished at: 2024-03-05T13:27:43+01:00
[INFO] ------------------------------------------------------------------------
```

So then it's not an issue with your sample project or the recipes, but hints at whatever is in that bash script or the `-Drewrite.recipeArtifactCoordinates=dev.morphia.morphia:rewrite:3.0.0-SNAPSHOT` it resolves, or the dependencies that comes with. 🤔  Not quite sure yet which of those, but perhaps this is already helping you in some way.

**Reply by timtebeek** [reply]
And know that we also have [a Slack you're welcome to join](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA) if that helps.
